jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/NameEnricherTest.java

```diff
@@ -105,4 +102,49 @@ public boolean matches(Object value) {
         .extracting(ObjectMeta::getName)
         .containsExactly("custom-name", "custom-name", "custom-name", "custom-name");
   }
+
+  @Test
+  public void create_withAlreadyExistingName_shouldKeepExistingName() {
+    // Given
+    klb = new KubernetesListBuilder();
+    klb.addToItems(new ServiceBuilder().withNewMetadata().withName("existing-name").endMetadata().build());
+    // When
+    new NameEnricher(context).create(PlatformMode.kubernetes, klb);
+    // Then
+    final Condition<Object> nonNull = new Condition<Object>("Not Null") {
+      @Override
+      public boolean matches(Object value) {
+        return value != null;
+      }
+    };
+    assertThat(klb.build().getItems())
+        .hasSize(1)
+        .extracting(HasMetadata::getMetadata)
+        .filteredOn(nonNull)
+        .extracting(ObjectMeta::getName)
+        .containsExactly("existing-name");
```

Comment:
I think that what you probably want is 

assertThat(klb.build().getItems()).singleElement().extracting(HasMetadata::getMetadata).extracting(ObjectMeta::getName).isEqualTo()

or 

assertThat(klb.build().getItems()).singleElement().extracting("metadata.name").isEqualTo("existing-name")

or

assertThat(klb.build().getItems()).singleElement().hasFieldOrPropertyWithValue("metadata.name", "existing-name");

?